src/_pytest/python.py

```diff
@@ -250,21 +251,9 @@ def pytest_pycollect_makeitem(collector: "PyCollector", name: str, obj: object):
             return res
 
 
-class PyobjMixin:
+class PyobjMixin(nodes.Node):
```

Comment:
Given this now subclasses `Node`, it is no longer a mixin (IMHO), can we find a better name for it, perhaps `PythonObject`? Also include a docstring.

Reply:
it still is a mixin and used as one - adding the base class just gets the typing data straight and ensures the mro is correct

Reply:
My interpretation of mixins is that they must not inherit from other classes, as they just add a bunch of methods and don't introduce inheritance (seems to match [this description](https://en.wikipedia.org/wiki/Mixin) too). If the mixin is subclassing another class, it no longer is a mixin because it now introduces inheritance relationships: anybody subclassing `PyobjMixin` previously would not be part of the `Node` hierarchy, but now they do.

Reply:
Since python supports multiple inheritance and node is practically a hard dependency for the mixin,

I consider it as a mixin in the sense that we place it in a specific part of the mro to add behaviour

The article is more langue agnostic, when multiple inheritance is supported then i consider it entirely acceptable to explicitly depend on a otherwise only implied dependency

Reply:
OK, I'm still not convinced but no worries.